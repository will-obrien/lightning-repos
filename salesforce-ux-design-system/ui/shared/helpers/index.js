// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import omit from 'lodash.omit';
import range from 'lodash.range';
import reverse from 'lodash.reverse';
import startCase from 'lodash.startcase';
import times from 'lodash.times';
import truncate from 'lodash.truncate';
import uniqueIdLodash from 'lodash.uniqueid';
import upperFirst from 'lodash.upperfirst';

// SHIM Lodash because it caches in node_modules and generates id's that are always incrementing
const uniqueId = (() => {
  const PREFIXES = {};
  let idCounter = 0;
  const newCounter = prefix => {
    PREFIXES[prefix] = 0;
  };
  const incCounter = prefix => {
    PREFIXES[prefix] = PREFIXES[prefix] + 1;
    return PREFIXES[prefix];
  };
  const initCounterForPrefix = prefix =>
    PREFIXES[prefix] != null ? PREFIXES[prefix] : newCounter(prefix);
  const addToPrefix = prefix => {
    initCounterForPrefix(prefix);
    return prefix + incCounter(prefix);
  };
  return prefix => (prefix ? addToPrefix(prefix) : idCounter++);
})();

export default {
  omit,
  range,
  reverse,
  startCase,
  times,
  truncate,
  uniqueId,
  upperFirst
};
