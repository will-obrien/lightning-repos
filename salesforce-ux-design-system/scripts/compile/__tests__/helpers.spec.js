// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */

import path from 'path';

import paths from '../../helpers/paths';
import { chunkedEntry, manifest } from '../helpers';

const __paths = Object.assign({}, paths);

beforeAll(() => {
  paths.root = path.resolve(__paths.root, '__fixtures__');
  paths.ui = path.resolve(__paths.root, '__fixtures__/ui');
});

afterAll(() => {
  Object.assign(paths, __paths);
});

it('creates an entry', () => {
  expect.assertions(1);
  chunkedEntry.fork(
    e => {
      throw e;
    },
    generatedEntry => {
      expect(generatedEntry.toJS()).toMatchSnapshot();
    }
  );
});

it('creates a manifest', () => {
  expect.assertions(1);
  manifest.fork(
    e => {
      throw e;
    },
    result => {
      expect(result.toJS()).toMatchSnapshot();
    }
  );
});
