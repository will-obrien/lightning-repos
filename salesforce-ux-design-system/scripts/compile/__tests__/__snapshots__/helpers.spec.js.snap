// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a manifest 1`] = `
Object {
  "./ui/components/buttons/base/example.jsx": "__internal/chunked/showcase/ui/components/buttons/base/example.jsx.js",
  "./ui/components/buttons/docs.mdx": "__internal/chunked/docs/ui/components/buttons/docs.mdx.js",
  "./ui/components/buttons/stateful/example.jsx": "__internal/chunked/showcase/ui/components/buttons/stateful/example.jsx.js",
  "./ui/components/buttons/with-icon/example.jsx": "__internal/chunked/showcase/ui/components/buttons/with-icon/example.jsx.js",
  "./ui/components/data-tables/advanced/example.jsx": "__internal/chunked/showcase/ui/components/data-tables/advanced/example.jsx.js",
  "./ui/components/data-tables/base/example.jsx": "__internal/chunked/showcase/ui/components/data-tables/base/example.jsx.js",
  "./ui/components/data-tables/inline-edit/example.jsx": "__internal/chunked/showcase/ui/components/data-tables/inline-edit/example.jsx.js",
  "./ui/utilities/alignment/example.jsx": "__internal/chunked/showcase/ui/utilities/alignment/example.jsx.js",
  "./ui/utilities/floats/example.jsx": "__internal/chunked/showcase/ui/utilities/floats/example.jsx.js",
  "docs/common": "__internal/chunked/docs/common.js",
  "showcase/common": "__internal/chunked/showcase/common.js",
}
`;

exports[`creates an entry 1`] = `
Object {
  "__internal/chunked/docs": Object {
    "__internal/chunked/docs/ui/components/buttons/docs.mdx.js": "./ui/components/buttons/docs.mdx",
  },
  "__internal/chunked/showcase": Object {
    "__internal/chunked/showcase/ui/components/buttons/base/example.jsx.js": "./ui/components/buttons/base/example.jsx",
    "__internal/chunked/showcase/ui/components/buttons/stateful/example.jsx.js": "./ui/components/buttons/stateful/example.jsx",
    "__internal/chunked/showcase/ui/components/buttons/with-icon/example.jsx.js": "./ui/components/buttons/with-icon/example.jsx",
    "__internal/chunked/showcase/ui/components/data-tables/advanced/example.jsx.js": "./ui/components/data-tables/advanced/example.jsx",
    "__internal/chunked/showcase/ui/components/data-tables/base/example.jsx.js": "./ui/components/data-tables/base/example.jsx",
    "__internal/chunked/showcase/ui/components/data-tables/inline-edit/example.jsx.js": "./ui/components/data-tables/inline-edit/example.jsx",
    "__internal/chunked/showcase/ui/utilities/alignment/example.jsx.js": "./ui/utilities/alignment/example.jsx",
    "__internal/chunked/showcase/ui/utilities/floats/example.jsx.js": "./ui/utilities/floats/example.jsx",
  },
}
`;
